community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/pipes/ExecuteUpdateCommandsPipe.scala

```diff
@@ -69,10 +69,7 @@ case class ExecuteUpdateCommandsPipe(source: Pipe, commands: Seq[UpdateAction])(
 
   def sourceSymbols: SymbolTable = source.symbols
 
-  override def localEffects = source match {
-    case pipe: ExecuteUpdateCommandsPipe => pipe.source.localEffects
-    case _ => commands.effects(source.symbols)
-  }
+  override def localEffects = commands.effects(sourceSymbols)
```

Comment:
If we had a test for that now it is green and happy. I am unsure what kind of problem that `match` was trying to fix.

On the other hand, it looks to me wrong that the `localEffects` of the pipe are the one of its source in case the source is another `ExecuteUpdateCommandsPipe` and ignoring completely the local commands.